### PR TITLE
136 add documentation for basic controller implementation

### PR DESCRIPTION
**Pull request by samvanderzwan**
Also fixed the building on read the docs so that autoclass is working and the docstring documentation is added on read the docs.

### CODE REVIEW

File: doc/controller/controller.rst [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
+lower then the available capacity of the observed sources, the remaining demand will be equally
+distributed over the source at the observed priority. In the case the demand is higher then the
+available source capacity, a message is passed to the user and the demand is downscaled to match
+the available capacity.
+
+When an storage is present in the network, the controller will first allocate the capacity of the
+sources to meet the demand of the consumers. The remaining capacity will be used to charge the storage.
+When the capacity of the sources is not sufficient to meet the demand of the consumers,
+the controller will use the storage to meet the remaining demand. This is a very simple
+implementation of a control strategy. In future more complex strategies might be added.
```
Comment (vanmeerkerk) [inline doc/controller/controller.rst:23]: Text needs to be adjusted for readability and conciseness:

The controller manages set points for assets in the simulator. ESDL data is parsed by mapper functions into controller objects, which are invoked at each time step to calculate and return the set points for that step. These set points are passed back to the simulator in a dictionary format, where the key is the asset ID, and the value is another dictionary. This inner dictionary holds the set points, with the keys being properties (e.g., supply temperature, heat demand) and the values being their corresponding set points.

The controller prioritizes source allocation based on a priority system. It first assigns capacity from priority 1 sources. If more capacity is needed, it moves to priority 2, and so on. If demand is lower than the available capacity at a given priority, the excess is equally distributed across the sources. If demand exceeds capacity, a message is sent to the user, and demand is downscaled to match available resources.

When storage is present, the controller first allocates source capacity to meet consumer demand. Any remaining capacity is used to charge the storage. If source capacity is insufficient, the controller taps into the storage to meet the remaining demand. This basic control strategy may be extended with more complex strategies in the future.

File: doc/controller/main_controller_class.rst [hunk 2]
```diff
@@ -2,10 +2,15 @@
 
 Network controller class
 =====================================
-The network controller class is hte basic controller class. It stores lists of the controllable
-assets. The method run_time_step is used to get the controller settings for the given time step.
-For developers if you want to implement a new controller a class can be created with its own logic
-as long as there is a method run_time_step that returns the controller settings.
+The network controller class is the basic controller class. It stores lists of the controllable
```
Comment (vanmeerkerk) [inline doc/controller/main_controller_class.rst:5]: The NetworkController class serves as the base controller, managing lists of controllable assets. The update_setpoints method retrieves controller settings for a specific time step.

To create a custom controller, implement a new class with your desired logic and inherit from the NetworkControllerAbstract class.

File: doc/requirements.txt [hunk 3]
```diff
@@ -3,4 +3,4 @@ sphinx-rtd-theme>=1.0.0
 sphinxcontrib-bibtex>=2.4.2
 python_docs_theme
 furo
-omotes-simulator-core
\ No newline at end of file
+.
```
Comment (vanmeerkerk) [inline doc/requirements.txt:6]: why do you use the dot?

File: pyproject.toml [hunk 4]
```diff
@@ -75,9 +75,14 @@ count_commits_from_version_file = true
 dev_template = "{tag}.dev{ccount}"
 dirty_template = "{tag}.dev{ccount}"
 
+[tool.setuptools]
```
Comment (vanmeerkerk) [inline pyproject.toml:78]: Why do you need these changes?

**Review by vanmeerkerk — COMMENTED** [review]
See comments

File: doc/requirements.txt (hunk 3 above)
Comment (samvanderzwan) [reply doc/requirements.txt:6]: To install the omotos-simulator-core

Comment (samvanderzwan) [reply doc/requirements.txt:6]: This is needed to make autodoc work

File: pyproject.toml (hunk 4 above)
Comment (samvanderzwan) [reply pyproject.toml:78]: Needed to correctly install the package when you run pip install . and I need to be able to do this to create the documentation on read the docs

File: doc/controller/controller.rst (hunk 1 above)
Comment (samvanderzwan) [reply doc/controller/controller.rst:23]: adjusted

File: doc/controller/main_controller_class.rst (hunk 2 above)
Comment (samvanderzwan) [reply doc/controller/main_controller_class.rst:5]: done

File: doc/requirements.txt (hunk 3 above)
Comment (vanmeerkerk) [reply doc/requirements.txt:6]: Ok it is the requirement file for the documentation. I missed that part of the change.

File: pyproject.toml (hunk 4 above)
Comment (vanmeerkerk) [reply pyproject.toml:78]: I still find this change weird, but lets commit the pr.

**Review by vanmeerkerk — APPROVED** [review]
Approve changes, with the comment that I find the change to the pyproject toml interesting/weird and I don't know whether it is necessary.

**Comment by vanmeerkerk** [comment]
@samvanderzwan something is going wrong with the closing of this commit. Can you check?

**Pull request merged**